src/Console/Infrastructure/Command/ListModulesCommand.php

```diff
@@ -25,55 +26,62 @@ protected function configure(): void
         $this->setName('list:modules')
             ->setDescription('Render all modules found')
             ->addArgument('filter', InputArgument::OPTIONAL, 'Any filter to simplify the output')
-            ->addOption('simple', 's', InputOption::VALUE_NONE, 'Display just the module names');
+            ->addOption('detailed', 'd', InputOption::VALUE_NONE, 'Display all the modules in detail');
     }
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $filter = (string)$input->getArgument('filter');
 
-        $listOfModules = $this->generateListOfModules(
-            (bool)$input->getOption('simple'),
+        $this->generateListOfModules(
+            $output,
+            (bool)$input->getOption('detailed'),
             $this->getFacade()->findAllAppModules($filter),
         );
 
-        $output->write($listOfModules);
-
         return self::SUCCESS;
     }
 
     /**
      * @param list<AppModule> $modules
      */
-    private function generateListOfModules(bool $isSimple, array $modules): string
+    private function generateListOfModules(OutputInterface $output, bool $isDetailed, array $modules): void
     {
-        return ($isSimple)
-            ? $this->generateSimpleView($modules)
-            : $this->generateDetailedView($modules);
+        if ($isDetailed) {
+            $this->generateDetailedView($output, $modules);
+            return;
+        }
+
+        $this->generateSimpleView($output, $modules);
```

Comment:
What about doing this instead?
```php
  ($isDetailed)
    ? $this->generateDetailedView($modules)
    : $this->generateSimpleView($modules);
```

Reply:
I think `else` in this scenario fits better, it is more explicit than your suggestion 😄